getter.go

```diff
@@ -119,7 +154,60 @@ func (g *getter) retryRequest(method, urlStr string, body io.ReadSeeker) (resp *
 	return
 }
 
-func (g *getter) initChunks() {
+func (bg *batchGetter) queueFile(path string) error {
+	url, err := bg.b.url(path, bg.c)
+
+	resp, err := bg.retryRequest("GET", url.String(), nil)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("Bad status for HTTP response: %d", resp.StatusCode)
+	}
+
+	// Golang changes content-length to -1 when chunked transfer encoding / EOF close response detected
+	if resp.ContentLength == -1 {
+		return fmt.Errorf("Retrieving objects with undefined content-length " +
+			" responses (chunked transfer encoding / EOF close) is not supported")
+	}
+
+	logger.debugPrintf("object size: %3.2g MB", float64(resp.ContentLength)/float64((1*mb)))
+	bg.initChunks(resp, path)
+	return nil
+}
+
+func (bg *batchGetter) initChunks(resp *http.Response, path string) {
+	id := 0
```

Comment:
This path variable refers to where to write the file to right?  It feels we can combine these single / batch getter modes where we provide the overall getter an optional WriteTo interface.   The only difference between the single and bath seems to be the path to write to